test/unit/lib/plugins/aws/provider.test.js

```diff
@@ -326,6 +327,28 @@ describe('AwsProvider', () => {
       return expect(awsProvider.options['aws-s3-accelerate']).to.be.undefined;
     });
   });
+
+  describe('#s3CodeArtifactsDirectoryPath', () => {
+    beforeEach(() => {
+      awsProvider.hooks.initialize();
+    });
+    it('should provide proper code artifacts S3 directory path', () => {
```

Comment:
Any new tests should be based on [`runServerless`](https://github.com/serverless/test/blob/master/docs/run-serverless.md) util, e.g. see how `provider.getRegion()` is tested: https://github.com/serverless/serverless/blob/9e308bdf041c73eb2d27408ae4e03bc10fe9bf8a/test/unit/lib/plugins/aws/provider.test.js#L602-L637